client/src/context/useSocketContext.tsx

```diff
@@ -1,18 +1,22 @@
-import { useState, useContext, createContext, FunctionComponent, useCallback } from 'react';
+import { useState, useContext, createContext, FunctionComponent, useCallback, useEffect } from 'react';
 import { io, Socket } from 'socket.io-client';
+import { Notification } from '../interface/User';
 
 interface ISocketContext {
   socket: Socket | undefined;
   initSocket: () => void;
+  socketNotification: Notification | undefined;
 }
 
 export const SocketContext = createContext<ISocketContext>({
   socket: undefined,
   initSocket: () => null,
+  socketNotification: undefined,
 });
 
 export const SocketProvider: FunctionComponent = ({ children }): JSX.Element => {
   const [socket, setSocket] = useState<Socket | undefined>(undefined);
+  const [socketNotification, setSocketNotification] = useState<Notification>();
```

Comment:
We already have a context for notifications, why are we storing them again here? :)

Reply:
This is the notification which is coming from the socket. I am storing it here and passing the 'socketNotification' to notificationContext. 
This was the only way I could get everything working.
I tried the other way around , getting the 'setNotifications' inside socketContext and updating the notifications. But that does not work. But this way everything functions.

Reply:
For future reference, please bear in mind that we want to have only one single point of truth (having a notifications state in two different places can lead to use of inaccurate data)

Reply:
Thank you., I will certainly keep that in mind and also will try to solve this ., whenever i get time.